packages/backend/server/src/core/doc-service/service.ts

```diff
@@ -0,0 +1,97 @@
+import { FactoryProvider, Injectable, Logger } from '@nestjs/common';
+import { ModuleRef } from '@nestjs/core';
+import { ClsService } from 'nestjs-cls';
+
+import {
+  Config,
+  CryptoHelper,
+  InternalServerError,
+  UserFriendlyError,
+} from '../../base';
+import { PgWorkspaceDocStorageAdapter } from '../doc';
+import { type DocRecord } from '../doc/storage';
+
+export abstract class DocService {
+  abstract getDoc(
+    workspaceId: string,
+    docId: string
+  ): Promise<DocRecord | null>;
+}
+
+@Injectable()
+class DatabaseDocService extends DocService {
+  constructor(protected readonly workspace: PgWorkspaceDocStorageAdapter) {
+    super();
+  }
+
+  async getDoc(workspaceId: string, docId: string): Promise<DocRecord | null> {
+    return await this.workspace.getDoc(workspaceId, docId);
+  }
+}
+
+@Injectable()
+class RpcDocService extends DocService {
+  private readonly logger = new Logger(DocService.name);
+
+  constructor(
+    private readonly config: Config,
+    private readonly crypto: CryptoHelper,
+    private readonly cls: ClsService
+  ) {
+    super();
+  }
+
+  async getDoc(workspaceId: string, docId: string): Promise<DocRecord | null> {
+    const url = `${this.config.docService.endpoint}/rpc/workspaces/${workspaceId}/docs/${docId}`;
+    const accessToken = `${docId},${this.crypto.sign(docId)}`;
+    try {
+      const response = await fetch(url, {
+        headers: {
+          'x-access-token': accessToken,
+          'x-rpc-trace-id': this.cls.getId(),
```

Comment:
After deployment, it didn't take effect. I need to print more logs to check

Reply:
it work 

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/hTwOityLamd4hitrae7M/c77f1295-bf16-4312-bfd5-50f6c1ba6c30.png)